.github/workflows/cypress-tests-runner.yml

```diff
@@ -22,6 +22,46 @@ env:
   RUST_MIN_STACK: 10485760
 
 jobs:
+  formatter:
+    name: Run formatter on Cypress tests and address lints
+    if: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref || github.ref }}
+
+      - name: Install Node.js
+        if: ${{ env.RUN_TESTS == 'true' }}
```

Comment:
I don't think you need this `if` condition since you've added the check on the job level.